### PR TITLE
Support running tests once (i.e. without the scheduler) against CODE environment

### PR DESCRIPTION
**Pull request by cb372**
Now you can run `sbt test:run` to run a subset of the tests (any tests that only make sense on PROD are excluded) once, instead of running them periodically using the Quartz scheduler.

This can be run on TeamCity and is also useful for devs to run on their local machines.

Details:
* Added a new standalone test runner (`RunCODETests.scala`)
* Disable SSL certificate checking for tests that run against CODE env, because CODE uses self-signed certificates
* Only enable PagerDuty alerts when running tests in the scheduler
* Replaced the package object with a few traits, plus various other refactoring
* Introduced a new `SanityTestBase` abstract class that mixes in a bunch of useful traits
* Introduced a new `@ProdOnly` annotation to mark a suite as only making sense on PROD
* Add a dependency on scalatestplus-play so we can use `OneAppPerSuite` to run a fake Play app

@gidsg @guardian/content-platforms 

It's a huge diff, so it might be easier to review it one commit at a time.

### CODE REVIEW

File: test/com/gu/contentapi/sanity/RunCODETests.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import com.gu.contentapi.sanity.support.{DoNothing, FakeAppSupport}
+import play.api.test.Helpers.running
+
+/**
+ * Run all the tests that are suitable for execution against the CODE environment.
+ *
+ * Use `sbt test:run` to run this class.
+ *
+ */
+object RunCODETests extends App {
```
Comment (gidsg) [inline test/com/gu/contentapi/sanity/RunCODETests.scala]: You should set up a project in TeamCity to use your new command to run the `CODE` tests and check they pass.

**Comment by gidsg** [comment]
The readme requires a further update to indicate that tests should no longer be added to the `test` root and describing how tests should now be added.

**Comment by cb372** [comment]
After a bit of battling with ScalaTest and TeamCity, this is now ready for another review.

File: test/com/gu/contentapi/sanity/CODESuite.scala [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.gu.contentapi.sanity.support.{FakeAppSupport, DoNothing}
+import org.scalatest.Suites
+
+/**
+ * Nested suite of all the tests that should be run in the CODE environment
+ */
+class CODESuite extends Suites(
+  Seq(
+    new DraftR2ContentShouldAppearInPreviewTest,
+    new ReadComposerDraftInPreviewTest
```
Comment (gidsg) [inline test/com/gu/contentapi/sanity/CODESuite.scala]: I think you need to update the README to reflect that CODE tests should be added to this class?

**Comment by gidsg** [comment]
:+1: once comment is addressed. :swimmer:

**Pull request merged**